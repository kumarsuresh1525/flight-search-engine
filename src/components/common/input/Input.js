import React, {useState, memo} from 'react';
import './Input.scss';

const Input = ({type, onChange, className, placeholder, isMandatory, min = 1, max = 100, defaultValue}) => {
  const [value, setValue] = useState(defaultValue);
  const [error, setError] = useState('');
  const handleChange = (event) => {
    const {value: inputValue} = event.target;
    setValue(inputValue);
    onChange(inputValue);
  }

  const handleBlur = () => {value.length < 1 && setError('No of passengers is mandatory')}
  return (
    <div>
      <input
        data-testid='input'
        type={type}
        className={className}
        onChange={handleChange}
        onBlur={handleBlur}
        value={value}
        min={min}
        max={max}
        placeholder={placeholder}
      />
      {isMandatory && <div className='error'>{error}</div>}
    </div>
  );
}

export default memo(Input);