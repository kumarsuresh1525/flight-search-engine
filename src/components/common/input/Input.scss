@import '../../../styles/global';

.input-number {
  font-size: 14px;
  height: 30px;
  margin-top: 20px;
  width: -webkit-fill-available;
}