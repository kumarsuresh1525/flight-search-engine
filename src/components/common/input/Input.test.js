import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Input from './Input';

describe('Input test cases', () => {
  const defaultProps = {
    filters: {},
    onChange: jest.fn(),
    className: 'className',
    placeholder: 'placeholder',
    defaultValue: '',
    min: 0,
    max: 100,
    type: 'range',
    isMandatory: true
  }

  test('renders Input correctly', () => {
    const { asFragment } = render(<Input {...defaultProps}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should be able to change the input', () => {
    const {getByTestId} = render(<Input {...defaultProps}/>);
    const input = getByTestId('input');
    fireEvent.change(input, {target: {value: 1}});
    expect(input.value).toBe('1');
  });

  it('Should be able show error on focus out if value is null', () => {
    const props = {...defaultProps, min: ''};
    const {getByTestId, getByText} = render(<Input {...props}/>);
    const input = getByTestId('input');
    fireEvent.blur(input);
    expect(getByText(/No of passengers is mandatory/)).toBeInTheDocument()
  });
});
