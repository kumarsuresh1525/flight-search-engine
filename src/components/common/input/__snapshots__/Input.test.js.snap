// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test cases renders Input correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      class="className"
      data-testid="input"
      max="100"
      min="0"
      placeholder="placeholder"
      type="range"
      value=""
    />
    <div
      class="error"
    />
  </div>
</DocumentFragment>
`;
